test/utils/cache.js

```diff
@@ -0,0 +1,41 @@
+'use strict'
+
+const { describe, test } = require('node:test')
+const assert = require('node:assert')
+const { parseVaryHeader } = require('../../lib/util/cache.js')
+
+describe('parseVaryHeader', () => {
```

Comment:
nit but could you move these to where the other tests are for this function? https://github.com/nodejs/undici/blob/a9176c9ec67b3efa1732799d18162c3a4307164e/test/cache-interceptor/utils.js#L179